src/ra_server_proc.erl

```diff
@@ -862,7 +863,11 @@ terminate(Reason, StateName,
 
         _ -> ok
     end,
-    _ = ets:delete(ra_metrics, Key),
+    case ServerState of
```

Comment:
Did you mean to simply destructure here or is the key optional?

Reply:
Key is mandatory I suppose. Destructure in function signature is better, yes. Thanks @michaelklishin !